compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/tree/SyntaxTreeModifierTest.java

```diff
@@ -109,6 +115,54 @@ public void testBinaryExprModification() {
         Assert.assertEquals(minusOrSlashCounter.transform(newRoot), new Integer(4));
     }
 
+    @Test
+    public void testSeparatedListNodeNonSeperatorModification() {
+        SyntaxTree syntaxTree = parseFile("separated_node_list_modify_all_nodes.bal");
+        ModulePartNode oldRoot = syntaxTree.rootNode();
+
+        ModuleVariableDeclarationNode oldModuleVariableDeclarationNode =
+                (ModuleVariableDeclarationNode) oldRoot.members().get(0);
+        ListConstructorExpressionNode oldSeperatedlistNode =
+                (ListConstructorExpressionNode) oldModuleVariableDeclarationNode.initializer();
+
+        IdentifierModifier identifierModifier = new IdentifierModifier();
+        ModulePartNode newRoot = (ModulePartNode) oldRoot.apply(identifierModifier);
+
+        ModuleVariableDeclarationNode newModuleVariableDeclarationNode =
+                (ModuleVariableDeclarationNode) newRoot.members().get(0);
+        ListConstructorExpressionNode newSeperatedlistNode =
+                (ListConstructorExpressionNode) newModuleVariableDeclarationNode.initializer();
+
+        Assert.assertEquals(oldSeperatedlistNode.expressions().separatorSize(),
+                newSeperatedlistNode.expressions().separatorSize());
+        Assert.assertEquals(oldSeperatedlistNode.expressions().size(),
+                newSeperatedlistNode.expressions().size());
+    }
+
+    @Test
+    public void testSeparatedListNodeAllNodeModification() {
+        SyntaxTree syntaxTree = parseFile("separated_node_list_modify.bal");
+        ModulePartNode oldRoot = syntaxTree.rootNode();
+
+        ModuleVariableDeclarationNode oldModuleVariableDeclarationNode =
+                (ModuleVariableDeclarationNode) oldRoot.members().get(0);
+        ListConstructorExpressionNode oldSeperatedlistNode =
+                (ListConstructorExpressionNode) oldModuleVariableDeclarationNode.initializer();
+
+        TokenModifier tokenModifier = new TokenModifier();
+        ModulePartNode newRoot = (ModulePartNode) oldRoot.apply(tokenModifier);
+
+        ModuleVariableDeclarationNode newModuleVariableDeclarationNode =
+                (ModuleVariableDeclarationNode) newRoot.members().get(0);
+        ListConstructorExpressionNode newSeperatedlistNode =
+                (ListConstructorExpressionNode) newModuleVariableDeclarationNode.initializer();
+
+        Assert.assertEquals(oldSeperatedlistNode.expressions().separatorSize(),
+                newSeperatedlistNode.expressions().separatorSize());
+        Assert.assertEquals(oldSeperatedlistNode.expressions().size(),
+                newSeperatedlistNode.expressions().size());
```

Comment:
These two asserts only checks for the counts.
We need to assert the content as well, to see whether they were updated correctly or not.

Reply:
ok